File: yht0827/2월/2월 25일/배달.java
class Solution {
	public int solution(int N, int[][] road, int K) {
		int answer = 0;

		int[][] arr = new int[N][N];

		for (int i = 0; i < N; i++) {
			for (int j = 0; j < N; j++) {
				if (i == j) {
					continue;
				}
				arr[i][j] = 500001;
			}
		}

		for (int i = 0; i < road.length; i++) {
			if (arr[road[i][0] - 1][road[i][1] - 1] >= road[i][2]) {
				arr[road[i][0] - 1][road[i][1] - 1] = road[i][2];
				arr[road[i][1] - 1][road[i][0] - 1] = road[i][2];
			}
		}

		for (int i = 0; i < N; i++) {
			for (int j = 0; j < N; j++) {
				for (int k = 0; k < N; k++) {
					if (arr[j][k] > arr[j][i] + arr[i][k]) {
						arr[j][k] = arr[j][i] + arr[i][k];
					}
				}
			}
		}

		for (int i = 0; i < N; i++) {
			if (arr[0][i] <= K) {
				answer++;
			}
		}
		return answer;
	}
}

File: jeong-hyuk/2월25일/배달.py
import heapq

def dijkstra(graph, start, K):
    distances = {node: float('inf') for node in graph}
    distances[start] = 0
    queue = []
    heapq.heappush(queue, [distances[start], start])

    while queue:
        current_distance, current_node = heapq.heappop(queue)

        if distances[current_node] < current_distance:
            continue

        for adjacent, weight in graph[current_node].items():
            distance = current_distance + weight

            if distance <= K and distance < distances[adjacent]:
                distances[adjacent] = distance
                heapq.heappush(queue, [distance, adjacent])

    count = 0
    for distance in distances.values():
        if distance <= K:
            count += 1

    return count

def solution(N, road, K):
    graph = {node: {} for node in range(1, N + 1)}

    for u, v, w in road:
        if v in graph[u]:
            graph[u][v] = min(graph[u][v], w)
            graph[v][u] = min(graph[v][u], w)
        else:
            graph[u][v] = w
            graph[v][u] = w

    return dijkstra(graph, 1, K)
